src/async_reader.rs

```diff
@@ -63,14 +63,22 @@ impl<B: AsyncBackend + Sync + Send, C: DirectoryCache + Sync + Send> AsyncPmTile
 
     /// Fetches tile bytes from the archive.
     pub async fn get_tile(&self, z: u8, x: u64, y: u64) -> Option<Bytes> {
+        self.try_get_tile(z, x, y).await.ok()
+    }
+
+    /// Fetches tile bytes from the archive, exposing the underlying error
+    /// (if any).
+    pub async fn try_get_tile(&self, z: u8, x: u64, y: u64) -> PmtResult<Bytes> {
```

Comment:
TileNotFound is not an error - I think we should have this:

```suggestion
    pub async fn try_get_tile(&self, z: u8, x: u64, y: u64) -> PmtResult<Option<Bytes>> {
```

Reply:
Done.